snapshot/snapshot.go

```diff
@@ -127,13 +127,19 @@ func NewSnapshotter(ctx context.Context, root string, targetFs FileSystem, opts
 		return nil, err
 	}
 
-	return &snapshotter{
+	o := &snapshotter{
 		context:     ctx,
 		root:        root,
 		ms:          ms,
 		asyncRemove: config.asyncRemove,
 		fs:          targetFs,
-	}, nil
+	}
+
+	if err := o.restoreRemoteSnapshot(ctx); err != nil {
+		return nil, errors.Wrap(err, "failed to restore remote snapshot")
+	}
```

Comment:
We should have a field on `config.toml` for enabling this feature.

Reply:
Since the service restart will cause the original snapshot to not work, I think the restore function should always be enabled